### PR TITLE
UI:Updates packages that have vulnerability

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes
    This patch updates vulnerable packages and enforces following changes
    
    - Modifies few tests and Resources component
    - Adds prettier rule for endOfLine as auto to resolve prettier errors
    - Moves global variable declaration into a `.d.ts` file
    to resolve eslint errors
    - Adds `.eslintcache` in the gitignore which is generated by
    latest version of react-scripts

Signed-off-by:Shiv Verma <shverma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/config/constants.tsx [hunk 1]
```diff
@@ -3,7 +3,9 @@ interface API_CONFIG {
   GH_CLIENT_ID: string;
 }
 
+// eslint-disable-next-line
```
Comment (PuneetPunamiya) [inline ui/src/config/constants.tsx:6]: Do we need this, I think we should not allow eslint disable

File: ui/src/config/constants.tsx [hunk 2]
```diff
@@ -3,7 +3,9 @@ interface API_CONFIG {
...
 declare global {
+  // eslint-disable-next-line
```
Comment (PuneetPunamiya) [inline ui/src/config/constants.tsx:8]: Same as above

File: ui/src/containers/Authentication/Authentication.test.tsx [hunk 3]
```diff
@@ -30,7 +30,8 @@ describe('Authentication', () => {
   });
 
   it('it can test github login', () => {
-    const windowOpenSpy = spyOn(window, 'open');
+    window.open = jest.fn();
```
Comment (PuneetPunamiya) [inline ui/src/containers/Authentication/Authentication.test.tsx:33]: Is this addition required as we are updating the packages ??

Comment (pratap0007) [reply ui/src/containers/Authentication/Authentication.test.tsx:33]: yeah , test was failing because of react-script@4.0.1 have a changeLog in jest

**Comment by vinamra28** [comment]
@pratap0007 can we have the commit description as well highlighting all the changes in this commit apart from commit message?
